### PR TITLE
Revision game_info & spreadsheet

### PR DESCRIPTION
**Pull request by yzemaze**
details s. commit comments
tr highlight on hover/tap limited to game_info & spreadsheet for now (only makes sense on larger tables, definitely not on .card)

### CODE REVIEW

File: assets/app/view/game/bank.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def render_card
           h('div.title.nowrap', title_props, [h(:em, 'The Bank')]),
           h(:div, body_props, [
             h(:div, @game.format_currency(@game.bank.cash)),
-            h(GameInfo, game: @game, layout: 'discarded_trains'),
+            *h(GameInfo, game: @game, layout: 'discarded_trains'),
```
Comment (tobymao) [inline assets/app/view/game/bank.rb:46]: render should always return one item, so remove the * and make sure GameInfo returns one div

File: assets/app/view/game/game_info.rb [hunk 2]
```diff
@@ -14,71 +16,41 @@ def render
...
-            discarded_trains,
-          ])
-        end
-
-        children << phases
-        children << game_info
-
-        h(:div, children)
+        children = upcoming_trains
+        children.concat discarded_trains unless @depot.discarded.empty?
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:25]: always use parenthesis

Comment (tobymao) [reply assets/app/view/game/game_info.rb:25]: prefer positive conditionals

if @depot.discarded.any?

File: assets/app/view/game/game_info.rb [hunk 3]
```diff
@@ -88,54 +60,51 @@ def phases
           end
 
           h(:tr, [
-            h(:td, td_props, phase[:name] + (current_phase == phase ? ' (Current) ' : '')),
-            h(:td, td_props, phase[:operating_rounds]),
-            h(:td, td_props, phase[:train_limit]),
+            h(:td, (current_phase == phase ? '⮕ ' : '') + phase[:name]),
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:63]: unicode looks inconsistent in browsers, what about -> or >

File: assets/app/view/game/game_info.rb [hunk 4]
```diff
@@ -147,62 +116,66 @@ def upcoming_trains
         rows = @depot.upcoming.group_by(&:name).map do |name, trains|
           train = trains.first
           discounts = train.discount&.group_by { |_k, v| v }&.map do |price, price_discounts|
-            price_discounts.map(&:first).join(',') + ' => ' + @game.format_currency(price)
+            price_discounts.map(&:first).join(', ') + ' ⮕ ' + @game.format_currency(price)
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:119]: same here

File: assets/app/view/game/game_info.rb (hunk 3 above)
Comment (dfannius) [reply assets/app/view/game/game_info.rb:63]: It would be a crime to represent an arrow by the ASCII string `->` in the year 2020.

Comment (tobymao) [reply assets/app/view/game/game_info.rb:63]: lol ok maybe the arrow is ok or possible you can make an svg

**Comment by tobymao** [comment]
can you change "Operating Order" to be "Order"

File: assets/app/view/game/bank.rb (hunk 1 above)
Comment (yzemaze) [reply assets/app/view/game/bank.rb:46]: It already returns `''` if there are no discarded trains. I was avoiding the div-wrapper around header + table on purpose as these are exactly the ones I tried to eradicate. I’ll find another solution.

File: assets/app/view/game/game_info.rb (hunk 2 above)
Comment (yzemaze) [reply assets/app/view/game/game_info.rb:25]: *notetoself* don’t copy anyone’s code blindly

File: assets/app/view/game/game_info.rb (hunk 3 above)
Comment (yzemaze) [reply assets/app/view/game/game_info.rb:63]: Arrows for consistency with spreadsheet (up/down arrow already indicates sort order there).
Even if it gets rendered slightly differently on different browsers it should still be clearly identifiable as an arrow ;) svg would just be more load.

File: assets/app/view/game/game_info.rb (hunk 4 above)
Comment (yzemaze) [reply assets/app/view/game/game_info.rb:119]: > can you change "Operating Order" to be "Order"

Sure. I’ll forward any hate-mail from spreadsheets grognards to you ;)

File: assets/app/view/game/game_info.rb (hunk 3 above)
Comment (yzemaze) [reply assets/app/view/game/game_info.rb:63]: Real question: Is the arrow sufficient to indicate current round? Additionally highlight row with background color maybe?

**Comment by tobymao** [comment]
i think an arrow is sufficient

**Pull request merged**